2_process/src/combine_NLCD_PRMS.R

```diff
@@ -0,0 +1,64 @@
+###-----
+
+## Subset Land to AOI and join with PRMS ids and COMID area (sqkm)
+AOI_LC_w_area <- function(NLCD_LC_df, area_att, aoi_comids_df){
+  
+  #' @description following the comid id as the main key, this function subset NLCD LC data to the area of interest, joins with nhd COMID area info and PRMS info
+  #' Output is a df with PRMS ids, COMIDs, AREA(SQKM), and NLCD LC percentage per COMID (unique key)
+  #' @param area_att df with comids and associated area (AREASQKM)
+  #' @param NLCD_LC_df selected LC dataframe structured by COMID 
+  #' @param aoi_comids_df df linking comid and prms for the area of interest, already filter to area of interest 
+  #' @example AOI_LC_w_area(NLCD_LC_df = p1_NLCD_df$NLCD_LandCover_2011, area_att = p2_nhd_area_att, drb_comids_df = p2_drb_comids)
+  
+# Filter land cover attributes to aoi comids (drb) and join with area columns
+  lc_area <- NLCD_LC_df %>%
+    filter(COMID %in% aoi_comids_df$comid) %>%
+    left_join(., area_att, by="COMID") %>%
+    mutate(COMID = as.character(COMID))
+
+# Create df with PRMS, COMID, and NLCD LC att for the aoi (drb)
+  lc_aoi <- aoi_comids_df %>%
+    left_join(., lc_area, by = c('comid' = "COMID"))
+
+  return(lc_aoi)
+
+  }
+
+###----
+# Estimate proportion of LC coverage at PRMS scale
+proportion_lc_by_prms <- function(NLCD_LC_df_w_area,
+                          catchment_att = "CAT"){
+
+#' @description This function takes the consolidated df from AOI_LC_w_area() and calculated the percentage share of each LC class at the PRMS segid scale
+#' @param NLCD_LC_df_w_area df output of AOI_LC_w_area that has LC data per COMID & PRMS id and area in (km2)
+#' @param catchment_att Specifies the catchment attribute to subset to. Must be either 'CAT', "ACC", 'TOT'. Default = "CAT"
+#' @example proportion_lc_by_prms(NLCD_LC_df_w_area = p2_LC_w_catchment_area, catchment_att = "CAT")
```

Comment:
I like the idea of allowing the user to specify "CAT", "ACC", or "TOT", but I'm wondering whether the "ACC" or "TOT" options are meaningful here. For example, if I specified "TOT" I would still sum only the direct catchment areas (given by `AREASQKM`) but then would be multiplying that area by proportion of land cover in the total upstream watershed (`TOTDASQKM`). For total upstream land cover I think we'll need a function that handles land cover differently (joins based on `comid_down` in the xwalk table and then directly choose either "ACC" vs "TOT" attribute values). That could be a separate PR, however, I suggest we omit the upstream choices from this function and rename the function/target to indicate that we're calculating catchment-scale attributes. What do you think? Do you agree with my interpretation above?

Reply:
Agreed. Removed this param